examples/scripts/project/categorization/bootstrap_taxonomy.py

```diff
@@ -0,0 +1,169 @@
+    category_list = []
+    for category in categories:
+        category_list.append(category.path)
+    # For bootstrapping at a specific tier = -1 (leaf), 1, 2, ... use the existing function
+    # category_set = metrics._get_categories_at_tier(project=project, tier=1)
+    # category_list = [category.split("|") for category in category_set]
+    category_list.sort()
+
+    # Create a dataset with taxonomy categories
+    dataset_name = f"{project.unified_dataset().name}_taxonomy_bootstrap_dataset"
+    try:
+        project.client.datasets.by_name(dataset_name)
+    except KeyError:
+        # Dataset with `dataset_name` does not exist in Tamr. Proceed with dataset creation.
+        pass
+    else:
+        dataset_exists_error = (
+            f"A dataset with name {dataset_name} already exists. Try again after deleting the "
+            "dataset."
+        )
+        LOGGER.error(dataset_exists_error)
+        raise RuntimeError(dataset_exists_error)
+
+    # Create a dictionary of full path between string and list (used as label path)
+    taxonomy_dict = {", ".join(category): category for category in category_list}
+
+    # Create a dataframe
+    df = pd.DataFrame(list(taxonomy_dict.items()), columns=COLUMN_NAMES)
+
+    # Add category data and hash for tamr_id
+    df[CATEGORY_ATTRIBUTE_NAME] = df[FULL_PATH_LIST_NAME].apply(lambda x: x[-1])
+    df[PK_NAME] = df[FULL_PATH_STR_NAME].apply(lambda x: hash(x))
+    df.drop(FULL_PATH_LIST_NAME, axis=1, inplace=True)
+
+    # Create a dataset in Tamr
+    taxonomy_dataset = project.client.datasets.create_from_dataframe(
+        df, primary_key_name=PK_NAME, dataset_name=dataset_name
+    )
+    LOGGER.info(f"Created a dataset in Tamr with name: {taxonomy_dataset.name}")
+
+    # Add the dataset into the project
+    project.add_input_dataset(taxonomy_dataset)
+    LOGGER.info(f"Added {taxonomy_dataset.name} to project {project.name}")
+
+    # Create a unified attribute to map taxonomy categories
+    unified_attribute_name = "Taxonomy Categories"
+    assert schema.create_unified_attribute(project, unified_attribute_name=unified_attribute_name)
+
+    # Set ML configuration for the newly created unified attribute
+    assert schema.set_unified_attribute_configurations(
+        project,
+        unified_attribute_name=unified_attribute_name,
+        similarity_function="COSINE",
+        tokenizer="DEFAULT",
+        attribute_role="",
+        is_numeric=False,
+    )
+    LOGGER.info(f"Created a unified attribute {unified_attribute_name}")
+
+    # Tamr_id management is needed for label insert
+    all_tx = tbox.project.schema_mapping.transformations.get_all(project)
+    new_input_tx = tbox.project.categorization.transformations.InputTransformation(
+        "SELECT *, to_string(origin_entity_id) AS tamr_id;", [taxonomy_dataset]
+    )
+    all_tx.input_scope.insert(0, new_input_tx)
+    tbox.project.schema_mapping.transformations.set_all(project, all_tx)
```

Comment:
The tamr_id should be a globally unique ID. If for example the the dataset was created twice in this way on the same instance then this would be violated. Instead, you should retrieve the tamr_id from the unified dataset after updating below

Reply:
Wouldn't streaming the unified dataset be inefficient for this? I would think that bootstrapping the second time should overwrite the results from the first time.

Reply:
Yes, it's inefficient but it's the only guaranteed correct way to reference the tamr_id AFAIK. It may be fine though if you also include the origin_source_name as part of the tamr_id